qiskit_nature/algorithms/pes_samplers/bopes_sampler.py

```diff
@@ -36,7 +38,7 @@ class BOPESSampler:
 
     def __init__(
         self,
-        gss: GroundStateSolver,
+        gss: Union[GroundStateSolver, ExcitedStatesSolver],
```

Comment:
In the new version, I opted for the name "solver_wrapper" which seemed appropriate

Reply:
We noticed now, that changing the naming here does not work as CI will fail when running the stable tutorials (because the tutorials used to specify this argument via a keyword.

@manoelmarques @woodsp-ibm what do you think on this matter? I would indeed like to update the name... Can we add a patch to the stable tutorials to use this argument positionally, to ensure that they do not break by such a change? We are going to update them once we release 0.4 anyways..

Reply:
I am fine with changing the stable tutorials for this as it will still work with released code and 0.4 is near anyway.
If you agree, I will create a main PR changing the main tutorials with a backport option.

Reply:
Nothing should break by doing this. If you want to rename it then gss should be kept as deprecated parameter, maybe add at the end, so if people did it by keyword it will not break. Hence the tutorials should be updated to avoid a deprecated parameter but they should not break if we can at all avoid it. Which I think we can in this case right.

Reply:
#688 addresses this and, once backported, you should be able to rename the first argument of this class as you had originally intended, @Anthony-Gandon

Reply:
Okay, we had some more offline discussion. Here is, what the strategy which we should follow:

1. Change `gss -> solver_wrapper=None` and add `gss=None` as a keyword argument at the end.
This ensures that everything works the same right now.
At the same time, you add a deprecation for the `gss` argument.

2. In the future, once we remove `gss`, we can also remove the default `None` value of the `solver_wrapper`, because in Python even a positional argument can be specified as a keyword argument later

@Anthony-Gandon if any of this is unclear, feel free to ping me here or offline :+1:

Reply:
So I was not sure how to deprecate the gss parameter. The latest commit:

- Adds a `@deprecate_arguments` to the constructor of BOPESSolver.
- Adds the gss argument at the end of the constructor for BOPES and sets the gss and solver_wrapper argument as `Optional`.
- Modifies the doctoring of the constructor
- Modifies the release note to add the deprecation.

Please let me know if that is the right way / if this is enough, thanks !

Reply:
(`solver_wrapper` was changed to `state_solver` following a later comment)